src/polycap-description.c

```diff
@@ -258,17 +262,177 @@ polycap_profile* polycap_description_get_profile(polycap_description *descriptio
+			} while(boundary_check == -1);
+		}
+	
+		// Obtain point from source as photon origin, determining photon start_direction
+		r = polycap_rng_uniform(rng);
+		src_rad_x = description->src_x * sqrt(fabs(r)); ////sqrt to simulate source intensity distribution (originally src_x * r/sqrt(r) )
+		r = polycap_rng_uniform(rng);
+		src_rad_y = description->src_y * sqrt(fabs(r)); ////sqrt to simulate source intensity distribution
+		r = polycap_rng_uniform(rng);
+		phi = 2.0*M_PI*fabs(r);
+		src_start_x = src_rad_x * cos(phi) + description->src_shiftx;
+		src_start_y = src_rad_y * sin(phi) + description->src_shifty;
+		if((description->src_sigx * description->src_sigy) < 1.e-20){ //uniform distribution over PC entrance
+			r = polycap_rng_uniform(rng);
+			pc_rad = description->profile->ext[0] * sqrt(fabs(r));
+			r = polycap_rng_uniform(rng);
+			phi = 2.0*M_PI*fabs(r);
+			pc_x = pc_rad * cos(phi) + start_coords.x;
+			pc_y = pc_rad * sin(phi) + start_coords.y;
+			start_direction.x = pc_x - src_start_x;
+			start_direction.y = pc_y - src_start_y;
+			start_direction.z = description->d_source;
+		} else { //non-uniform distribution, direction vector is within +- sigx
+			r = polycap_rng_uniform(rng);
+			start_direction.x = description->src_sigx * (1.-2.*fabs(r));
+			r = polycap_rng_uniform(rng);
+			start_direction.y = description->src_sigy * (1.-2.*fabs(r));
+			start_direction.z = 1.;
+		}
+		polycap_norm(&start_direction);
 
+		// Create photon structure
+		photon = polycap_photon_new(rng, start_coords, start_direction, start_electric_vector, n_energies, energies);
 
+		// Launch photon
+		istart++; //Here all photons that started, also enter the polycapillary
+		photon->i_refl = 0; //set reflections to 0
+		iesc = polycap_photon_launch(photon, description);
+			//if iesc == -1 here a new photon should be simulated/started.
+			//	essentially do j-1 as this will have same effect
+		if(iesc == -1) j--;
 
+		//COUNT function here... If required...
+
+		if(thread_id == 0 && (double)i/((double)icount/(double)thread_cnt/10.) >= 1.){
+			printf("%d%%\t%" PRId64 "\t%f\n",((j*100)/(icount/thread_cnt)),photon->i_refl,photon->exit_coords.z);
+			i=0;
+		}
+		i++;//counter just to follow % completed
+
+		//save photon->weight in thread unique array
+		for(k=0; k<n_energies; k++) weights[k] += photon->weight[k];
+		#pragma omp critical
+		{
+		sum_irefl += photon->i_refl;
+		}
+
+		//free photon structure (new one created for each for loop instance)
+		polycap_photon_free(photon);
+
+	} //for(j=0; j < icount; j++)
+
+	#pragma omp critical
+	{
+	sum_istart += istart;
+	for(i=0; i<n_energies; i++) sum_weights[i] += weights[i];
+	}
+	polycap_rng_free(rng);
+	free(weights);
+} //#pragma omp parallel
+
+	printf("Average number of reflections: %f\n",(double)sum_irefl/5000.);
+
+	for(i=0; i<n_energies; i++){
+		*efficiencies[i] = (sum_weights[i] / sum_istart) * description->open_area;
```

Comment:
is dit de juiste manier om **efficiencies te dereferencen en de waarden aan te passen? Double pointers en al...

Reply:
Nope, want de gebruiker zal een referentie doorgeven naar een niet-gealloceerde pointer.

Voor uw for loop moet ge dus eerst gaan alloceren:

````c
*efficiencies = malloc(sizeof(double) * n_energies);
````

Daarna zou het moeten werken: tijd voor een unit-test 😄 

Mooi werk trouwens de voorbije dagen! 👍